sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -241,7 +241,7 @@ case class NewInstance(
         $javaType ${ev.value} = ${ctx.defaultValue(dataType)};
         if ($argsNonNull) {
           ${ev.value} = $constructorCall;
-          ${ev.isNull} = false;
+          ${ev.isNull} = ${ev.value} == null;
```

Comment:
Actually it's a irrelevant change and I forgot to revert it before push commits.
The reason I made this change is: if we think a constructor may return null, why we skip the null check in `propagateNull = true` branch? If @marmbrus think this change makes sense, we should also set the `nullalble` to true.